src/routes.rs

```diff
@@ -112,11 +116,40 @@ fn refresh_token(origin: Option<cors::Origin>,
                  auth_param: AuthParam,
                  configuration: State<Configuration>,
                  cors_options: State<TokenGetterCorsOptions>,
+                 keys: State<Keys>,
                  authenticator: State<Box<auth::BasicAuthenticator>>)
                  -> Result<cors::Response<Token<PrivateClaim>>, ::Error> {
 
+    if !configuration.refresh_token_enabled() {
+        return Err(::Error::BadRequest("Refresh token is not enabled".to_string()));
+    }
+    let refresh_token_configuration = configuration.refresh_token();
+
     auth_param.verify(&authorization)?;
-    Err(::Error::UnsupportedOperation)
+    let refresh_token = RefreshToken::new_encrypted(&authorization.token());
```

Comment:
The token was received in an encrypted form. This is calling the enum's "constuctor" to create the encrypted variant. See https://github.com/lawliet89/rowdy/pull/23/files#diff-8eaf0fafad7d2ed9ec9a701c04424c96R412